### PR TITLE
Karpenter version v0.15.0 support

### LINKED ISSUE #5562: [Task] Update Karpenter to v0.13.2
### Description

Atm, we have Karpenter versioned pinned to 0.9.0. Upgrade this version to [0.13.2](https://github.com/aws/karpenter/releases/tag/v0.13.2)

Previous Reference ticket: #5023 

Edit 18-08-2022:
AWS notified us of a new stable release of v0.15.0 yesterday so we'll be updating the minimum supported version to `0.15.0`

### PR DESCRIPTION
**Pull request by Himangini**
### Description
Updated supported karpenter version to v0.15.0
Also updated docs 

Closes #5562 

Successfully manually tested following the same steps mentioned [here](https://github.com/weaveworks/eksctl/pull/5201)

Create cluster with karpenter v0.13.2
```
$ eksctl create cluster -f examples/29-cluster-with-karpenter.yaml
...
2022-08-15 17:30:46 [ℹ]  using region us-west-2
2022-08-15 17:30:47 [ℹ]  setting availability zones to [us-west-2b us-west-2a us-west-2c]
....
2022-08-15 17:51:50 [ℹ]  adding Karpenter to cluster cluster-with-karpenter-gini
2022-08-15 17:52:18 [ℹ]  kubectl command should work with "/Users/himangini/.kube/config", try 'kubectl get nodes'
2022-08-15 17:52:18 [✔]  EKS cluster "cluster-with-karpenter-gini" in "us-west-2" region is ready
```

Create cluster with Karpenter 0.15.0
```
$ ./eksctl create cluster -f examples/29-cluster-with-karpenter.yaml
2022-08-18 12:29:44 [ℹ]  eksctl version 0.110.0-dev+554ee903f.2022-08-18T12:16:50Z
2022-08-18 12:29:44 [ℹ]  using region us-west-2
2022-08-18 12:29:45 [ℹ]  setting availability zones to [us-west-2c us-west-2a us-west-2d]
...
2022-08-18 12:50:26 [ℹ]  adding Karpenter to cluster cluster-with-karpenter
2022-08-18 12:51:00 [ℹ]  kubectl command should work with "/Users/himangini/.kube/config", try 'kubectl get nodes'
2022-08-18 12:51:00 [✔]  EKS cluster "cluster-with-karpenter" in "us-west-2" region is ready

```

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by TiberiuGC — APPROVED** [review]

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -160,7 +160,8 @@ func validateKarpenterConfig(cfg *ClusterConfig) error {
 		return fmt.Errorf("failed to parse karpenter version %q: %w", cfg.Karpenter.Version, err)
 	}
 
-	if minor := v.Segments()[1]; minor > supportedKarpenterVersionMinor {
+	supportedVersion, err := version.NewVersion(supportedKarpenterVersion)
+	if err != nil || v.GreaterThan(supportedVersion) {
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:164]: The error must be handled explicitly here and propagated to the caller.

Comment (Himangini) [reply pkg/apis/eksctl.io/v1alpha5/validation.go]: done 👍🏻

**Review by cPu1 — APPROVED** [review]

**Comment by Pitta** [comment]
Holy Moley thank you!

**Comment by Pitta** [comment]
Does this being merged mean it is in the latest release? I didn't see it in todays release notes.

**Comment by Pitta** [comment]
Went ahead and built the dev binary to test - seems to have taken both k8s version `1.23` AND karpenter `0.15.0`!

**Comment by Himangini** [comment]
> Does this being merged mean it is in the latest release? I didn't see it in todays release notes.

We haven't had a chance to release this due to some internal issues. We're working on them and will aim to release a new RC soon. Thanks for your patience.

**Pull request merged**